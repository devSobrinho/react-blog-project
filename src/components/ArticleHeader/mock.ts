import { ArticleHeaderProps } from '.';
import mockArticleMeta from '../ArticleMeta/mock';
import mockHeader from '../Header/mock';

const header = {
  ...mockHeader,
  showText: false,
};

export default {
  header: header,
  title: 'Ipsum movet liberavisse an mea, diceret vituperata',
  excerpt:
    'Abhorreant reformidans eum an, per equidem repudiandae no, ex cum elitr argumentum persequeris.',
  cover:
    'https://res.cloudinary.com/dlizakp2a/image/upload/v1614696630/Frozen_sunset_on_the_lake_by_Manuel_Arslanyan_9f9cd8ea10.jpg',
  articleMeta: mockArticleMeta,
} as ArticleHeaderProps;
