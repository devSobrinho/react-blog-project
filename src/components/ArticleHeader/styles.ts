import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    margin: 0 auto;
    max-width: ${theme.sizes.max};
    padding-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    & ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      /* color: ${theme.colors.darkText}; */
      font-size: ${theme.font.sizes.huge};
    }

    @media${theme.media.lteMedium} {
      & ${HeadingStyles} {
        font-size: ${theme.font.sizes.xlarge};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: 2.4rem;
    margin: ${theme.spacings.medium} 0;
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
