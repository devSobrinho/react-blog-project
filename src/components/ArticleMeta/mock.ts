import { ArticleMetaProps } from '.';

export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'daniel-sobrinho',
    displayName: 'Daniel Sobrinho',
  },
  categories: [
    {
      id: 'nnfgr3n434232',
      displayName: 'React',
      slug: 'react',
    },
    {
      id: 'a33d32d32fafa',
      displayName: 'JS',
      slug: 'javascript',
    },
  ],
} as ArticleMetaProps;
