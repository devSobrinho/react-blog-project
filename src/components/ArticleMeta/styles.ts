import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkGray};
    font-style: italic;

    .categories span::after {
      content: ', ';
    }
    .categories span:last-child::after {
      content: '';
    }

    p {
      font-size: ${theme.font.sizes.small};
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }

    @media${theme.media.lteMedium} {
      p {
        font-size: ${theme.font.sizes.small};
      }
    }
  `}
`;
