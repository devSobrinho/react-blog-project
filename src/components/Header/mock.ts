import { HeaderProps } from '.';

export default {
  blogName: 'Daniel Sobrinho',
  blogDescription:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud.',
  showText: true,
  logo: {
    url: 'assets/images/dev-sobrinho.jpeg',
    id: 'qqqq11w1',
    alternativeText: 'textoalte',
  },
} as HeaderProps;
