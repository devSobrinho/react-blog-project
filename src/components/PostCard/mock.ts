import { PostCardProps } from '.';
import { data } from '../../api/dados.json';

export default data.posts[0] as PostCardProps;
