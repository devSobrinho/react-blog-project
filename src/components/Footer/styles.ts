import styled, { css } from 'styled-components';
import { Container as HtmlContent } from '../HtmlContent/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${HtmlContent} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
