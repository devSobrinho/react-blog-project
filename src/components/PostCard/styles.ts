import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;

    ${HeadingStyles} {
      margin-top: calc(${theme.spacings.small} - 0.6rem);
      margin-bottom: calc(${theme.spacings.small} - 0.6rem);
    }

    ${HeadingStyles}:hover {
      color: ${theme.colors.primary};
    }

    a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }

    &:hover ${Cover} {
      opacity: 0.8;
    }
    &:hover a {
      color: ${theme.colors.secondary};
    }
  `}
`;

export const Cover = styled.img`
  max-width: 100%;
  transition: all 300ms ease-in-out;
`;

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
  `}
`;
