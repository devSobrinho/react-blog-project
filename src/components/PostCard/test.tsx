import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PostCard, PostCardProps } from '.';
import mock from './mock';

const props: PostCardProps = mock;

describe('<PostCard />', () => {
  it('should render a heading, cover and excerpt', () => {
    renderTheme(<PostCard {...props} />);

    expect(screen.getByRole('link', { name: 'post-photo' })).toHaveAttribute(
      'href',
      `/post/${props.slug}`,
    );
    expect(
      screen.getByRole('img', { name: props.cover.alternativeText }),
    ).toHaveAttribute('src', props.cover.url);
    expect(
      screen.getByRole('heading', { name: props.title }).parentElement,
    ).toHaveStyleRule('display', 'flex');
    expect(
      screen.getByRole('heading', { name: props.title }),
    ).toBeInTheDocument();
    expect(screen.getByRole('link', { name: props.title })).toHaveAttribute(
      'href',
      `/post/${props.slug}`,
    );
    expect(screen.getByText(props.excerpt)).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<PostCard {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
