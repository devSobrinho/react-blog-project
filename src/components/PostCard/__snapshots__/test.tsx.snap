// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
.c5 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c0 {
  width: 100%;
}

.c0 .c4 {
  margin-top: calc(1.6rem - 0.6rem);
  margin-bottom: calc(1.6rem - 0.6rem);
}

.c0 .c4:hover {
  color: #000000;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover .c1 {
  opacity: 0.8;
}

.c0:hover a {
  color: #dc143c;
}

.c2 {
  max-width: 100%;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
>
  <a
    aria-label="post-photo"
    href="/post/teste"
  >
    <img
      alt=""
      class="c1 c2"
      src="https://res.cloudinary.com/dhgenbrm2/image/upload/v1643225348/0bbc3d47bac364c50b42c4b810fe602c7ef6f2bf_1_197a38aca9.jpg"
    />
  </a>
  <div
    class="c3"
  >
    <h2
      class="c4 c5"
    >
      <a
        href="/post/teste"
      >
        All the Lorem Ipsum generators on the Internet tend to repeat
      </a>
    </h2>
    <p
      class=""
    >
      ard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, a
    </p>
  </div>
</div>
`;
